routes/projects/files/create-update.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * PUT /projects/:username/:projectName/files/:pathToFile
+ * The :pathToFile parameter must be URI encoded
+ * Create or replace the contents for the given file path
+ * The message body that would normally be expected by this route
+ * will look something like:
+ * {
+ *   type: <string> content-type,
+ *   path: <string>,
+ *   data: <string> base64 encoded binary data
```

Comment:
It would be nice if this could be `String` (utf8) or `ArrayBuffer` (binary).  Having to deal in base64 for binary data sucks.